app/renderer/src/actions/Inspector.js

```diff
@@ -849,22 +879,72 @@ export function setAwaitingMjpegStream(isAwaiting) {
   };
 }
 
+export function uploadGesturesFromFile(fileList) {
+  return async (dispatch) => {
+    const fileReaderPromise = fileList.map((file) => {
+      const reader = new FileReader();
+      return new Promise((resolve) => {
+        reader.onload = (event) =>
+          resolve({
+            name: file.name,
+            content: event.target.result,
+          });
+        reader.readAsText(file);
+      });
+    });
+    const gestures = await Promise.all(fileReaderPromise);
+    const invalidGestures = [];
+    const parsedGestures = [];
+    gestures.forEach((gestureFile) => {
+      try {
+        const gesture = JSON.parse(gestureFile.content);
+        const isValid = gesture.name && gesture.actions.map(isValidGesture).every(Boolean);
+        if (!isValid) {
+          invalidGestures.push(gestureFile.name);
+        } else {
+          gesture.description = gesture.description || '';
+          parsedGestures.push(_.omit(gesture, ['id']));
+        }
+      } catch (e) {
+        invalidGestures.push(gestureFile.name);
+      }
+    });
+
+    if (invalidGestures.length) {
+      notification.error({
+        message: `Unable to upload gesture files ${invalidGestures.join(',')} due to invalid format`,
+      });
+    }
+    if (parsedGestures.length) {
+      await saveGesture(parsedGestures)(dispatch);
+      await getSavedGestures()(dispatch);
+    }
+  };
+}
+
 export function saveGesture(params) {
   return async (dispatch) => {
+    if (!_.isArray(params)) {
```

Comment:
I prefer to not reassign argument values. Consider creating another variable or adding the expression to the loop. Check https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern for more details